Scripts/Runtime/Data/VirtualData/VirtualData.cs

```diff
@@ -0,0 +1,296 @@
+
+            foreach (IVirtualData source in sources)
+            {
+                virtualData.AddSource(source);
+                source.AddResultDestination(virtualData);
+            }
+
+            return virtualData;
+        }
+
+        private readonly HashSet<IVirtualData> m_Sources = new HashSet<IVirtualData>();
+        private readonly HashSet<IVirtualData> m_ResultDestinations = new HashSet<IVirtualData>();
+        private readonly AccessController m_AccessController;
+
+        private UnsafeTypedStream<TInstance> m_Pending;
+        private DeferredNativeArray<TInstance> m_Iteration;
+        private UnsafeHashMap<TKey, TInstance> m_Lookup;
+
+        /// <summary>
+        /// The underlying <see cref="IDeferredNativeArray"/> used for scheduling
+        /// <see cref="IAnvilJobForDefer"/> or <see cref="IAnvilJobBatchDefer"/> jobs.
+        /// </summary>
+        public DeferredNativeArray<TInstance> ArrayForScheduling
+        {
+            get => m_Iteration;
+        }
+
+        AccessController IVirtualData.AccessController
+        {
+            get => m_AccessController;
+        }
+
+        private VirtualData()
+        {
+            m_AccessController = new AccessController();
+            m_Pending = new UnsafeTypedStream<TInstance>(Allocator.Persistent,
+                                                         Allocator.TempJob);
+            m_Iteration = new DeferredNativeArray<TInstance>(Allocator.Persistent,
+                                                             Allocator.TempJob);
+
+            m_Lookup = new UnsafeHashMap<TKey, TInstance>(MAX_ELEMENTS_PER_CHUNK, Allocator.Persistent);
+        }
+
+        protected override void DisposeSelf()
+        {
+            m_Pending.Dispose();
+            m_Iteration.Dispose();
+            m_Lookup.Dispose();
+
+            RemoveFromSources();
+            m_ResultDestinations.Clear();
+            m_Sources.Clear();
+
+            m_AccessController.Dispose();
+
+            base.DisposeSelf();
+        }
+
+        //*************************************************************************************************************
+        // SERIALIZATION
+        //*************************************************************************************************************
+
+        //TODO: Add Serialization and Deserialization functions to hook into our serialization framework
+
+        //*************************************************************************************************************
+        // RELATIONSHIPS
+        //*************************************************************************************************************
+
+        void IVirtualData.AddResultDestination(IVirtualData resultData)
+        {
+            AddResultDestination(resultData);
+        }
+
+        private void AddResultDestination(IVirtualData resultData)
+        {
+            m_ResultDestinations.Add(resultData);
+        }
```

Comment:
I don't understand. Why not just make this internal to satisfy the interface and omit the proxy method?

Same applies to the rest of times this happens

Reply:
Not allowed. 

https://stackoverflow.com/questions/68592353/why-cannot-implicitly-implement-a-non-public-interface-member

I need the interface because I need to store and operate on `VirtualData<TKey,TInstance>` as a collection. 

I suppose I could make an `AbstractVirtualData` instead that has the internal methods on it marked as abstract and then implement and seal them if you think that's better?